department-of-veterans-affairs/caseflow
# Allows tasks to block dispatch

Resolves #14056 

Part 1 of the stack:

Part 1: this PR
Part 2: #15041 
Part 3: #15042 

### Description
Renames 'blocking?' to 'blocking_distribution?'
Implements 'blocking_dispatch' and applies it to [appropriate tasks](https://github.com/department-of-veterans-affairs/caseflow/issues/14260#issue-616982555)


### Acceptance Criteria
- [ ] Full Stack: @lomky and @hschallhorn have doublechecked our enable reasoning
- [ ] Certain tasks will have the ability to block dispatch
   - [ ] CongressionalInterestMailTask
   - [ ] FoiaRequestMailTask
   - [ ] PrivacyActRequestMailTask
   - [ ] PrivacyComplaintMailTask
   - [ ] FoiaColocatedTask
   - [ ] FoiaTask
   - [ ] PrivacyActTask
- (Stack 2) These task should be created under RootTask* or BvaDispatchTask (Stack 2)
- (Stack 3) BvaDispatch tasks are not created when tasks that block dispatch are still open (Stack 3)
- (Stack 3) BvaDispatchTask is created when all tasks that block dispatch are closed (Stack 3)

### Testing Plan
1. Load up the console!
2. Create a direct review appeal at ready for distribution
3. Create Tasks of the types listed in the AC - the **should not** be created under the DistributionTask
   - [ ] Dispatch blocking tasks do not block Distribution
4. Create a few tasks that should still block distribution - the **should** be created under the Distribution Task
   - [ ] Distribution blocking tasks block Distribution


- Spend a bit of time grepping through the codebase with various permutations of the word 'blocking'
   - [ ] places that say 'blocking' now specify 'blocking distribution' or 'blocking dispatch'
   - [ ] places testing for blocking distribution properly invoke a task that still blocks distribution
